entoas/extension.go

```diff
@@ -0,0 +1,196 @@
+package entoas
+
+import (
+	"encoding/json"
+	"errors"
+	"io"
+	"os"
+	"path/filepath"
+
+	"entgo.io/contrib/entoas/policy"
+	"entgo.io/contrib/entoas/spec"
+	"entgo.io/ent/entc"
+	"entgo.io/ent/entc/gen"
+)
+
+type (
+	// Config provides global metadata for the generator. It is injected into the gen.Graph.
+	Config struct {
+		// DefaultPolicy defines the default policy for endpoint generation.
+		// It is used if no policy is set on a (sub-)resource.
+		// Defaults to PolicyExpose.
+		DefaultPolicy policy.Policy
+		// Whether or whether not to generate simple models instead of one model per endpoint.
```

Comment:
Maybe add a link to the spec.
https://spec.openapis.org/oas/v3.1.0#simple-model